### DISCUSSION TITLE
Merge a branch from another branch in master

### DISCUSSION THREAD

**Discussion by PierBJX** [community/community#21704] [New to GitHub]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/aa81fa88d927bf92208511ddb376b9373bb7e894.jpeg" data-download-href="/uploads/short-url/oknEaIRM3t3s2NEuariL63Jiz4M.jpg?dl=1" title="example.jpg" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/22519163/181066406-5bc75818-38dc-45cc-9643-59fdde03bdde.jpg" alt="example.jpg" data-base62-sha1="oknEaIRM3t3s2NEuariL63Jiz4M" width="690" height="214" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/aa81fa88d927bf92208511ddb376b9373bb7e894_2_690x214.jpg, https://aws1.discourse-cdn.com/github/optimized/1X/aa81fa88d927bf92208511ddb376b9373bb7e894_2_1035x321.jpg 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/aa81fa88d927bf92208511ddb376b9373bb7e894_2_1380x428.jpg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/aa81fa88d927bf92208511ddb376b9373bb7e894_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">example.jpg</span><span class="informations">3756×1169 541 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><strong>Following this picture:</strong></p>
<ul>
<li>I create a branch from master (Branch 1)</li>
<li>I push 3 commits.</li>
<li>I create a pull request on master.</li>
<li>Instead of waiting the branch 1 to be merged in master, I create a branch 2 from the branch 1.</li>
<li>I push 3 commits, meanwhile the branch 1 as been merged in master with the pull request.</li>
<li>I create a pull request for branch 2.</li>
</ul>
<p><strong>My questions are:</strong></p>
<ul>
<li>In my 2nd pull request, will l have only the 3 commits of the branch 2 or the 6 commits of the branch 1 and 2 ? Indeed, given that the branch 1 has been merged in master its 3 commits are no longer new in branch 2.</li>
<li>Same questions for the files changed in branch 1, will they appear in the pull request of the branch 2 even if the branch 1 has been merged in master ?</li>
</ul>
<p>Hope this is clear enought. Thanks</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/pierbjx">@pierbjx</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<blockquote>
<p><a class="mention" href="/pierbjx">@pierbjx</a> wrote:</p>
</blockquote>
<blockquote>
<p><strong>My questions are:</strong></p>
</blockquote>
<blockquote>
<ul>
<li>In my 2nd pull request, will l have only the 3 commits of the branch 2 or the 6 commits of the branch 1 and 2 ? Indeed, given that the branch 1 has been merged in master its 3 commits are no longer new in branch 2.</li>
</ul>
</blockquote>
<p>If the branch1 are already merge into master, the changes made by the 3 commits of branch1 are now also in master. The master already have the 3 commits of the branch1 then the Pull Request for branch2 analyze only the 3 new commits made by himself.</p>
<blockquote>
<p><a class="mention" href="/pierbjx">@pierbjx</a> wrote:</p>
</blockquote>
<blockquote>
<ul>
<li>Same questions for the files changed in branch 1, will they appear in the pull request of the branch 2 even if the branch 1 has been merged in master ?</li>
</ul>
</blockquote>
<p>Simple no, for the reason explain above.</p>
<p>Hope this helps, ask if you have more questions.</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>
